Update transport docs and settings for changes (#36786)
elastic/elasticsearch: docs/reference/modules/transport.asciidoc

as search in Elasticsearch.

[float]
=== Transport Settings

> The internal transport communicates over TCP. It allows for the following

Reviewer: ```suggestion
The internal transport communicates over TCP. You can configure it with the following
```
